src/rust/engine/process_execution/src/remote_cache.rs

```diff
@@ -555,12 +555,17 @@ async fn check_action_cache(
         .await
         .map_err(|e| Status::unavailable(format!("Output roots could not be loaded: {e}")))?;
 
-        check_action_cache_content(&response, store, cache_content_behavior)
+        let cache_content_valid = check_cache_content(&response, &store, cache_content_behavior)
           .await
           .map_err(|e| {
             Status::unavailable(format!("Output content could not be validated: {e}"))
           })?;
-        Ok(response)
+
+        if cache_content_valid {
+          Ok(response)
+        } else {
+          Err(Status::not_found(""))
```

Comment:
Emtpy string is fine?

Reply:
Yea: it's matched immediately below as `Code::NotFound`.